src/simulations/collect.jl

```diff
@@ -1,20 +1,12 @@
 export run!, collect_agent_data!, collect_model_data!,
        init_agent_dataframe, init_model_dataframe, aggname,
-       should_we_collect, Identity, Copy, DeepCopy
-
-abstract type Obtainer end
-struct Identity <: Obtainer end
-struct Copy <: Obtainer end
-struct DeepCopy <: Obtainer end
+       should_we_collect
 
 ###################################################
 # Definition of the data collection API
 ###################################################
-get_data(a, p, o::Type{O}) where {O<:Obtainer} = _get_data(a, p)
-get_data(a, p, o::Type{Copy}) = copy(_get_data(a, p))
-get_data(a, p, o::Type{DeepCopy}) = deepcopy(_get_data(a, p))
-_get_data(a, s::Symbol) = getproperty(a, s)
-_get_data(a, f::Function) = f(a)
+get_data(a, s::Symbol; obtainer = identity) = obtainer(getproperty(a, s))
+get_data(a, f::Function; obtainer = identity) = obtainer(f(a))
```

Comment:
For performance reasons, low-level functions that participate in hot loops should avoid the usage of keywords: `; -> ,`